### PR TITLE
Update listbox reducer with filter

### PR DESCRIPTION
**Pull request by pochretien**

### CODE REVIEW

**Comment by pochretien** [comment]
Simple change for the *Update* action with the listBox to keep the selected values.

Refactoring for Interface

File: src/ReactVapor.ts [hunk 1]
```diff
@@ -81,6 +81,10 @@ export interface IReactVaporState {
     lastAction?: Redux.Action;
 }
 
+export interface IReduxActionsDefaultPayload {
```
Comment (gdostie) [inline src/ReactVapor.ts]: BasePayload already exists somewhere in the repo

File: src/components/listBox/ListBoxReducers.ts [hunk 2]
```diff
@@ -79,21 +86,53 @@ export const listBoxReducer = (state: IListBoxState = listBoxInitialState, actio
...
+            const items: IItemBoxProps[] = _.map(state.items, (item: IItemBoxProps) => {
+                const hidden: boolean = _.some(state.selected, (selectedValue: string) => selectedValue === item.value);
+                return {
+                    ...item,
+                    hidden,
+                };
+            });
             return {
                 ...state,
+                items: items,
```
Comment (gdostie) [inline src/components/listBox/ListBoxReducers.ts]: `items,`

Comment (MichelML) [reply src/components/listBox/ListBoxReducers.ts]:
```
items: state.items((item) => ({...item, hidden: _.contains(state.selected, item.value)}))  
```  
you can remove the `const items` logic altogether if you just write the items property this way

**Pull request closed without merging**